test/integration/listener_lds_integration_test.cc

```diff
@@ -441,22 +442,25 @@ TEST_P(ListenerIntegrationTest, ChangeListenerAddress) {
   test_server_->waitUntilListenersReady();
   // NOTE: The line above doesn't tell you if listener is up and listening.
   test_server_->waitForCounterGe("listener_manager.listener_create_success", 1);
+  const uint32_t old_port = lookupPort(listener_name_);
   // Make a connection to the listener from version 1.
-  codec_client_ = makeHttpConnection(lookupPort(listener_name_));
+  codec_client_ = makeHttpConnection(old_port);
 
   // Change the listener address from loopback to wildcard.
-  const std::string new_address = ipVersion() == Network::Address::IpVersion::v4 ? "0.0.0.0" : "::";
+  const std::string new_address = Network::Test::getAnyAddressString(ipVersion());
   listener_config_.mutable_address()->mutable_socket_address()->set_address(new_address);
   sendLdsResponse({MessageUtil::getYamlStringFromMessage(listener_config_)}, "2");
   sendRdsResponse(fmt::format(route_config_tmpl, route_table_name_, "cluster_0"), "2");
 
   test_server_->waitForCounterGe("listener_manager.listener_create_success", 2);
   registerTestServerPorts({listener_name_});
+  const uint32_t new_port = lookupPort(listener_name_);
+  EXPECT_NE(old_port, new_port);
```

Comment:
Can you add a comment on why this is here? Something along the lines of "Want to verify that the port changed, because connecting to 127.0.0.1:80 works both for listener 127.0.0.1:80 and 0.0.0.0:80, so the port needs to be different to validate that the change in listener was actually made"